src/OrchardCore/OrchardCore/Modules/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -272,6 +273,71 @@ private static void AddAntiForgery(OrchardCoreBuilder builder)
             });
         }
 
+        /// <summary>
+        /// Adds backwards compatibility to the handling of SameSite cookies.
+        /// </summary>
+        public static void AddSameSiteCookieBackwardsCompatibility(OrchardCoreBuilder builder)
+        {
+            builder.ConfigureServices(services =>
+                {
+                    services.Configure<CookiePolicyOptions>(options =>
+                    {
+                        options.MinimumSameSitePolicy = AspNetCore.Http.SameSiteMode.Unspecified;
+                        options.OnAppendCookie = cookieContext => CheckSameSiteBackwardsCompatiblity(cookieContext.Context, cookieContext.CookieOptions);
+                        options.OnDeleteCookie = cookieContext => CheckSameSiteBackwardsCompatiblity(cookieContext.Context, cookieContext.CookieOptions);
+                    });
+                })
+                .Configure(app =>
```

Comment:
Should we assign an explicit order to ensure it's not executed too late? I don't remember what the default order is.

Reply:
It's application init logic, so it gets executed in the order in which it is called. And the order is really early.